### PR TITLE
✨ feat(#402): add login required component with custom redirect

### LINKED ISSUE #402: Make it possible to click "log in" in different places and redirect to same page
- If viewing an event without being logged in, the user should be able to click a button or link to log in (other than going profile -> log in). 
- When the user is logged in, they should be returned to the page the were at before logging in.

### PR DESCRIPTION
**Pull request by larwaa**
## Proposed Changes

- Add `<LoginRequired />`-component
- Add helper function for creating Feide logins with custom redirects
- Replace text on `EventDetails.tsx` with an interactive button

**Question**: Should we somehow escape whatever gets sent as state to Feide?

## Types of Changes

- [x] New feature
- [ ] Bug fix
- [x] Enhancement/optimization
- [ ] Refactor
- [ ] Style
- [ ] Build/dependencies
- [ ] Other

## Issue(s) fixed or closed by this PR

- Closes #402 

## Checklist

- [ ] I have added tests to cover my changes (for features/bugs)
- [x] I am pleased with the readability of the code (if not, state where you need input)
- [x] I have tested the changes and verified that they work and do not break anything (to the best of my ability)

### CODE REVIEW

File: frontend/src/components/navbar/NavbarLinks.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { GET_USER } from "@graphql/users/queries";
 import { User } from "@interfaces/users";
 import { Box, Button, makeStyles, Menu, MenuItem } from "@material-ui/core";
 import { AccountCircleOutlined, LockOpen } from "@material-ui/icons";
-import { DATAPORTEN_SCOPES } from "@utils/auth";
+import { DATAPORTEN_SCOPES, generateFeideLoginUrl } from "@utils/auth";
```
Comment (MorganFH) [inline frontend/src/components/navbar/NavbarLinks.tsx:7]:
```suggestion
import { generateFeideLoginUrl } from "@utils/auth";
```

File: frontend/src/utils/auth.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import { config } from "./config";
+import { generateQueryString } from "./helpers";
+
 export const DATAPORTEN_SCOPES = [
```
Comment (MorganFH) [inline frontend/src/utils/auth.ts:4]:
```suggestion
const DATAPORTEN_SCOPES = [
```

**Review by MorganFH — CHANGES_REQUESTED** [review]
Looks good!

If my suggestion above is accepted, can we also replace the now legacy log-in code in `frontend/src/pages/forms/[...formId].tsx`?

**Comment by larwaa** [comment]
> Looks good!
> 
> If my suggestion above is accepted, can we also replace the now legacy log-in code in `frontend/src/pages/forms/[...formId].tsx`?

Yep, that should be possible, great catch!

**Review by MorganFH — APPROVED** [review]
Very nice!

**Pull request merged**